src/components/Molecules/SearchBar/SearchBar.tsx

```diff
@@ -0,0 +1,126 @@
+import { useCallback, useState } from "react";
+import { Input } from "@/components/ui/input";
+import { useDebounce } from "@/utils/hooks/use-debounce";
+import { AddressActivityListView } from "@/components/Organisms/TokenBalances/AddressActivityListView/AddressActivityListView";
+import { TypographyH3 } from "@/components/ui/typography";
+import { ChainSelector } from "../ChainSelector/ChainSelector";
+import { useCovalent } from "@/utils/store/Covalent";
+import { BlockDetails } from "../BlockDetails/BlockDetails";
+import { type Chain } from "@covalenthq/client-sdk";
+import { TransactionReceiptView } from "@/components/Organisms/TransactionReceiptView/TransactionReceiptView";
+import { AddressDetailsView } from "@/components/Organisms/AddressDetailsView/AddressDetailsView";
+
+export const SearchBar: React.FC = () => {
+    const { selectedChain } = useCovalent();
+
+    const [searchInput, setSearchInput] = useState<string>("");
+    const [searchType, setSearchType] = useState<
+        | "address"
+        | "tx"
+        | "block"
+        // | "token"
+        | "not found"
+        | "none"
+    >("none");
+
+    useDebounce(
+        () => {
+            if (searchInput && selectedChain) {
+                handleSearch();
+            }
+        },
+        500,
+        [searchInput, selectedChain]
+    );
+
+    const handleSearch = useCallback(() => {
+        const addressRegex = /^0x[a-fA-F0-9]{40}$/;
+        const txnHashRegex = /^0x[a-fA-F0-9]{64}$/;
+        const blockRegex = /^\d+$/;
+        // const tokenRegex = /^[a-zA-Z0-9]+$/;
+        const domainNameRegex = /^[a-zA-Z0-9.-]+$/;
+
+        setSearchType("none");
+
+        if (addressRegex.test(searchInput)) {
+            setSearchType("address");
+        } else if (txnHashRegex.test(searchInput)) {
+            setSearchType("tx");
+        } else if (blockRegex.test(searchInput)) {
+            setSearchType("block");
+        }
+        // else if (tokenRegex.test(searchInput)) {
+        //     setSearchType("token");
+        // }
+        else if (domainNameRegex.test(searchInput)) {
+            setSearchType("address");
+        } else {
+            setSearchType("not found");
+        }
+    }, [searchInput]);
+
+    const handleResults = useCallback(() => {
+        switch (searchType) {
+            case "address": {
+                return (
+                    <div className="flex w-full flex-col gap-y-8">
+                        <AddressDetailsView
```

Comment:
I was thinking rather than rendering a component we can just return a string type (address, tx, block). This way the users will have the option to do whatever they want with this information ie. Routing somewhere else like what etherscan does or rendering their own custom components, etc

Reply:
We can do that, but it wont be an element anymore then, itll be a function - we can export a custom hook `useSearch` for it
what do you think?

Reply:
Yes good idea, could be a helper function that users could use for goldrush

Reply:
I also think keep the search bar but include the helper function in it so it shows the usecase and on how its usage is

Reply:
7dc4cd1f24b4cd51a67478a73ac1e2003037c0b8